fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -180,9 +180,122 @@ def additional_upload_parameters
     end
   end
 
+  # Generates commands and executes the altool.
+  class AltoolTransporterExecutor < TransporterExecutor
+    ERROR_REGEX = /\*\*\* Error:\s+(.+)/
+
+    private_constant :ERROR_REGEX
+
+    def execute(command, hide_output)
+      if Helper.test?
+        yield(nil) if block_given?
+        command
+      end
+
+      @errors = []
+      @all_lines = []
+
+      if hide_output
+        # Show a one time message instead
+        UI.success("Waiting for App Store Connect transporter to be finished.")
+        UI.success("Application Loader progress... this might take a few minutes...")
+      end
+
+      begin
+        exit_status = FastlaneCore::FastlanePty.spawn(command) do |command_stdout, command_stdin, pid|
+          command_stdout.each do |line|
+            @all_lines << line
+            parse_line(line, hide_output) # this is where the parsing happens
+          end
+        end
+      rescue => ex
+        # FastlanePty adds exit_status on to StandardError so every error will have a status code
+        exit_status = ex.exit_status
+        @errors << ex.to_s
+      end
+
+      @errors << "The call to the altool completed with a non-zero exit status: #{exit_status}. This indicates a failure." unless exit_status.zero?
+
+      unless @errors.empty? || @all_lines.empty?
+        # Print the last lines that appear after the last error from the logs
+        # If error text is not detected, it will be 20 lines
+        # This is key for non-verbose mode
+        error_line_index = @all_lines.rindex { |line| ERROR_REGEX.match?(line) }
+        @all_lines[(error_line_index || -20)..-1].each do |line|
+          UI.important("[altool] #{line}")
+        end
+        UI.message("Application Loader output above ^")
+        @errors.each { |error| UI.error(error) }
+      end
+
+      yield(@all_lines) if block_given?
+      exit_status.zero?
+    end
+
+    def build_upload_command(username, password, source = "/tmp", provider_short_name = "", jwt = nil, platform = nil, api_key = nil)
+      use_api_key = !api_key.nil?
+      [
+        ("API_PRIVATE_KEYS_DIR=#{Dir.tmpdir}" if use_api_key),
```

Comment:
Of course if you decide to go with my suggestion of using `TempFile` to generate the temporary p8 file, you'll have to adjust this to pass `File.dirname(tmp_p8_file_path)` there

Reply:
>  instead of passing the actual API key value 

Other API key value is necessary like `key_id` and `issuer_id`, so I wanna you this api_key argument as current actual API key value